docs/source/model_parallel.rst

```diff
@@ -0,0 +1,42 @@
+.. role:: hidden
+    :class: hidden-section
+
+Distributed RPC Framework
+=====================================================
+
+The distributed RPC framework provides mechanisms for multi-machine model training through a set of primitives to allow for remote communication, and a higher-level API to automatically differentiate models split across several machines.
+
+RPC and RRef Framework
+====================================================
+
+Basics
+------
+
+Before using RPC and distributed autograd primitives, initialization must take place. First, a backend over which RPCs can be sent over must be initialized. The default (and currently, only available) implementation is the `ProcessGroup` backend, and must be initialized with `torch.distributed.init_process_group` before using other function. See the `documentation for torch.distributed <https://pytorch.org/docs/stable/distributed.html>`_ for additional details. Next, the local RPC agent can be initialized, after which the process will be able to send and receive RPCs from all other connected processes.
+
+.. automodule:: torch.distributed.rpc
+    :members:
```

Comment:
@pritamdamania87 this will add all members of this module, but we'd want to be careful in using this since non-polished/private functions could show up.